src/main/java/com/artipie/SliceFromConfig.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.artipie;
+
+import com.artipie.composer.http.PhpComposer;
+import com.artipie.files.FilesSlice;
+import com.artipie.gem.GemSlice;
+import com.artipie.http.Response;
+import com.artipie.http.Slice;
+import com.artipie.http.async.AsyncSlice;
+import com.artipie.maven.http.MavenSlice;
+import com.artipie.npm.Npm;
+import com.artipie.npm.http.NpmSlice;
+import com.artipie.rpm.http.RpmSlice;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import org.cactoos.map.MapEntry;
+import org.cactoos.map.MapOf;
+import org.reactivestreams.Publisher;
+
+/**
+ * Slice from repo config.
+ * @since 0.1.4
+ * @todo #90:30min We still don't have tests for Pie. But now that this class was extracted, we have
+ *  a more cohesive class that could be tested. Write unit tests for SliceFromConfig class.
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class SliceFromConfig implements Slice {
+
+    /**
+     * Repository config.
+     */
+    private final RepoConfig config;
+
+    /**
+     * Ctor.
+     * @param config Repo config
+     */
+    public SliceFromConfig(final RepoConfig config) {
+        this.config = config;
+    }
+
+    @Override
+    public Response response(final String line, final Iterable<Map.Entry<String, String>> headers,
+        final Publisher<ByteBuffer> body) {
+        return new AsyncSlice(
```

Comment:
@HDouss instead of creating `AsyncSlice` instance in `response` method of `Slice` on each request you can use `Slice.Wrap` decorator for wrapping this logic:
```java
class SliceFromConfig extends Slice.Wrap {
  public SliceFromConfig(RepoConfig config) {
    super(
      new AsyncSlice(SliceFromConfig.build(config))
    );
  }
}
```

Reply:
@g4s8 I don't think it is a good idea. First, that will make the call to `SliceFromConfig.build` synchronous with the SliceFromConfig construction, so with Pie.response method. Second, for the requests coming from Pie, it is anyway building a new SliceFromConfig on each request, it is even building a new wrapping AsyncSlice.

Reply:
@HDouss but `SliceFromConfig.build(config)` is returning `CompletionStage` so it's not synchronous, right? Then actual creating of `Slice` will be deferred until next request. I didn't get your second point, but you are right: we're building new `Slice` for each request (we may add some caching later), it's correct.
The problem with current implementation is only in the design: you are creating new instance of same level in method, but it can be moved to constructor. Creating new concrete implementation in method increases class coupling, but constructor may accept interface types, so coupling will be low.

Reply:
@g4s8 What I meant is that the call to build is synchronous. But you are right, now with the last commit modifications it is not synchronous anymore. The second point is about the usage by Pie. Anyway fixed. Please see last commit.